codestodians/food-basket-api
# [Bug: #154304255] Fix previously unhandled errors/exceptions in user schema.

#### What does this PR do?
- It fixes previously unhandled possible errors/exceptions in the `user schema`.
- It ensures that the `email` is validated before allowing any mutation.

#### Description of Task to be completed?
- Update queries and mutations of the user schema.
- Write corresponding tests for new cases introduced.

#### How should this be manually tested?
- Run the server with `gunicorn config.wsgi`.
- Exhaustively test the operations considering different scenarios.
- Check that errors/exceptions are well handled.

#### Any background context you want to provide?
- This story arose as a result of running a query for a non-existent user on the `graphiql` Interface, discovering that we did not properly handle that particular case. Subsequently, other queries and mutations were scrutinized for possible errors.

#### Issue still needing a fix
- Only the user that owns an account should be allowed to update the details of that account, which means that possibly, an header value must be supplied for all operations except `createUser` and `user_login`. Haven't quite figured out the logic involved to make this happen. The goto would be to obtain the token of the `current_user` and check against the value supplied in the header, but I am still trying to ascertain how to check the header value in an operation. Once that is achieved, we could handle that in a separate story.

- A more elegant solution will be to make use of an `authorization middleware` or something along that line. The picture of how that will work is not quite formed in my mind yet though.

#### What are the relevant pivotal tracker stories?
- [#154304255](https://www.pivotaltracker.com/story/show/154304255)
